docs/codeql/codeql-for-visual-studio-code/running-codeql-queries-at-scale-with-mrva.rst

```diff
@@ -0,0 +1,184 @@
+:tocdepth: 1
+
+.. _running-codeql-queries-at-scale-with-mrva:
+
+Running CodeQL queries at scale with multi-repository variant analysis
+======================================================================
+
+.. include:: ../reusables/beta-note-mrva.rst
+
+About multi-repository variant analysis
+---------------------------------------
+
+When you write a query to find variants of a security vulnerability and finish testing it locally, the next step is to run it on a large group of repositories. Multi-repository variant analysis (variant analysis) makes it easy run a query on up to 1000 repositories without leaving Visual Studio Code.
+
+The core functionality of the CodeQL extension helps you write queries and run them locally against a CodeQL database. In contrast, variant analysis allows you to send your CodeQL query to GitHub.com to be tested against a list of repositories.
+
+When you run variant analysis against a list of repositories, your query is run against each repository that has a CodeQL database available to analyze. GitHub creates and stores CodeQL databases for thousands of public repositories, including every repository that runs code scanning using CodeQL. If you want to include your repositories in variant analysis, you need to enable code scanning using CodeQL on GitHub.com before adding your repository to a list for analysis. For information about enabling code scanning using CodeQL, see "`Configuring code scanning automatically <https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning-for-a-repository#configuring-code-scanning-automatically>`__."
+
+.. _controller-repository:
+
+About the controller repository
+-------------------------------
```

Comment:
I'd really like to reduce the entire section about the controller repository if possible as i think it could be quite distracting. Ultimately the user has a simple choice:

* make the controller repo public to analyze public codebases 
* make the controller repo private to analyze private codebases (different billing applies)

```suggestion
Setting a controller repository for variant analysis
-------------------------------------------------
```